src/reporter.ts

```diff
@@ -0,0 +1,164 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import type {
+  JUnitTestSuite,
+  JUnitTestCase,
+  Failure,
+  Error,
+  Property,
+  ConfigOption,
+} from './type';
+import { create } from 'xmlbuilder2';
+
+export class TestSuite implements JUnitTestSuite {
+  name: string;
+  tests: number;
+  failures: number;
+  errors: number;
+  time: number;
+  timestamp: string;
+  file: string;
+  testsuites: JUnitTestSuite[] | undefined;
+  testCases: JUnitTestCase[];
+  properties?: Property[] | undefined;
+
+  constructor() {
+    this.name = '';
+    this.time = 0;
+    this.tests = 0;
+    this.failures = 0;
+    this.errors = 0;
+    this.timestamp = '';
+    this.file = '';
+    this.testCases = [];
+    this.testsuites = [];
+  }
+
+  add(testcase: JUnitTestCase) {
+    this.testCases.push(testcase);
+  }
+}
+
+export class TestCase implements JUnitTestCase {
+  name: string;
+  classname: string;
+  time: number;
+  file: string;
+  failure?: Failure | undefined;
+  error?: Error | undefined;
+
+  constructor(name: string, classname: string, time: number, file: string) {
+    this.name = name;
+    this.classname = classname;
+    this.time = time;
+    this.file = file;
+  }
+}
+
+export default class Reporter {
+  public junitSuite: JUnitTestSuite;
+  private opts: ConfigOption;
+
+  constructor(opts: ConfigOption) {
+    this.junitSuite = new TestSuite();
+    this.opts = opts;
+  }
+
+  addTestSuite(suite: JUnitTestSuite) {
+    this.junitSuite.testsuites?.push(suite);
+  }
+
+  setFailures(failures: number) {
+    this.junitSuite.failures = failures;
+  }
+
+  setTests(tests: number) {
+    this.junitSuite.tests = tests;
+  }
+
+  setTime(time: number) {
+    this.junitSuite.time = time;
+  }
+
+  // Set the name of the root test suite.
```

Comment:
What about docs for the other methods? I added this doc here as an example, because it's necessary to know that setters here only apply to the root suite.

Reply:
👍